scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -573,14 +573,20 @@ def handle(self):
 
         except subprocess.CalledProcessError:
             os.write(write_pipe, b'x')  # halt the thread
+            self.instance.status = "error"
+            self.instance.reason = "Device issue (Flash error)"
+            flash_error = True
 
         if post_flash_script:
             self.run_custom_script(post_flash_script, 30)
 
         if not flash_error:
             t.join(self.timeout)
-            if t.is_alive():
-                logger.debug("Timed out while monitoring serial output on {}".format(self.instance.platform.name))
+        else:
```

Comment:
does this fix your issues? it seems that you just change the waiting time for serial monitor thread when met a flash error, hang up which point?

Reply:
just checked the twister log, the exception actually occurred at line 592 `ser.close()` as below, then the current process won't continue to execute and invoke later board release function `make_device_available`, so the board will always be  shown as unavailable, then other processes could not get an available device and show "waiting for device to become available" forever.
```
ile "/home/ztest/zephyrproject/zephyr/scripts/pylib/twister/twisterlib/runner.py", line 713, in run
    instance.handler.handle()
  File "/home/ztest/zephyrproject/zephyr/scripts/pylib/twister/twisterlib/handlers.py", line 586, in handle
    ser.close()
  File "/home/ztest/.local/lib/python3.8/site-packages/serial/serialposix.py", line 533, in close
    os.close(self.fd)
OSError: [Errno 9] Bad file descriptor
```

Reply:
Thank for the help on checking this issue together, @chen-png ! I add some comments on the code to clarify.